src/main/java/com/citusdata/migration/DynamoDBTableReplicator.java

```diff
@@ -69,567 +69,570 @@
+        } else if (typedValue.getBS() != null) {
+            List<ByteBuffer> value = typedValue.getBS();
+            return new TableColumnValue(TableColumnType.jsonb, Jackson.toJsonString(value));
+        } else if (typedValue.getL() != null) {
+            List<AttributeValue> value = typedValue.getL();
+            List<Object> simpleList = InternalUtils.toSimpleList(value);
+            return new TableColumnValue(TableColumnType.jsonb, Jackson.toJsonString(simpleList));
+        } else if (typedValue.getM() != null) {
+            Map<String, AttributeValue> value = typedValue.getM();
+            Item simpleMap = Item.fromMap(InternalUtils.toSimpleMapValue(value));
+            return new TableColumnValue(TableColumnType.jsonb, simpleMap.toJSON());
+        } else if (typedValue.getN() != null) {
+            if (convertNumberToText == true) {
+                String value = typedValue.getN();
+                return new TableColumnValue(TableColumnType.text, value);
+            } else {
+                String value = typedValue.getN();
+                return new TableColumnValue(TableColumnType.numeric, value);
+            }
+        } else if (typedValue.getNS() != null) {
+            List<String> value = typedValue.getNS();
+            return new TableColumnValue(TableColumnType.jsonb, Jackson.toJsonString(value));
+        } else if (typedValue.getS() != null) {
+            String value = typedValue.getS();
+            return new TableColumnValue(TableColumnType.text, value);
+        } else if (typedValue.getSS() != null) {
+            List<String> value = typedValue.getSS();
+            return new TableColumnValue(TableColumnType.text, Jackson.toJsonString(value));
+        } else if (typedValue.getNULL() != null) {
+            return new TableColumnValue(TableColumnType.text, "");
+        } else {
+            return null;
+        }
+    }
+
+    public static TableColumnType columnTypeFromDynamoValue(AttributeValue typedValue,  boolean convertNumberToText) {
+        if (typedValue.getB() != null) {
+            return TableColumnType.bytea;
+        } else if (typedValue.getBOOL() != null) {
+            return TableColumnType.bool;
+        } else if (typedValue.getBS() != null) {
+            return TableColumnType.jsonb;
+        } else if (typedValue.getL() != null) {
+            return TableColumnType.jsonb;
+        } else if (typedValue.getM() != null) {
+            return TableColumnType.jsonb;
+        } else if (typedValue.getN() != null) {
+            if (convertNumberToText == true) {
+                return TableColumnType.text;
+            } else
+                return TableColumnType.numeric;
+        } else if (typedValue.getNS() != null) {
+            return TableColumnType.jsonb;
+        } else if (typedValue.getS() != null) {
+            return TableColumnType.text;
+        } else if (typedValue.getSS() != null) {
+            return TableColumnType.jsonb;
```

Comment:
PR is longer open, but if someone sometime will use it please think about:
In columnValueFromDynamoValue TableColumnType.text is used for StringSet (SS) value, but column type is jsonb. So podyn will add jsonb column, but try to write text inside. In our case this lead to empty values and some debugging time.

Reply:
Want me to close this PR? This solved my issue